pkg/watcher/reconciler/pipelinerun/reconciler.go

```diff
@@ -92,7 +94,11 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, pr *pipelinev1.PipelineR
 	dyn.IsReadyForDeletionFunc = r.areAllUnderlyingTaskRunsReadyForDeletion
 	dyn.AfterDeletion = func(ctx context.Context, object results.Object) error {
 		pr := object.(*pipelinev1.PipelineRun)
-		return r.metrics.DurationAndCountDeleted(ctx, r.configStore.Load().Metrics, pr)
+		return r.pipelineRunMetrics.DurationAndCountDeleted(ctx, r.configStore.Load().Metrics, pr)
+	}
+	dyn.AfterStorage = func(ctx context.Context, object results.Object, _ bool) error {
+		pr := object.(*pipelinev1.PipelineRun)
```

Comment:
I see we do the same type assertion above, but we can improve on it. As written, if the assertion fails, it'll result runtime error and I believe the watcher will crash. instead we should check with `pr, ok := object.(*pipelinev1.PipelineRun)` and handle the !ok case more gracefully.
We have the same pattern also below in multiple places and we should improve on them as well.

Reply:
Sure, makes sense.